packages/grid/_modules_/grid/models/gridSlotsComponent.ts

```diff
@@ -66,6 +67,10 @@ export interface GridSlotsComponent extends GridIconSlotsComponent {
    * No rows overlay component rendered when the grid has no rows.
    */
   NoRowsOverlay?: React.ElementType;
+  /**
+   * All filtered overlay component rendered when the grid has no results after filtering.
+   */
+  AllFilteredOverlay?: React.ElementType;
```

Comment:
I'm not a big fan of my initial name proposal. Do we have better candidates?

Reply:
I have some: NoFilterResultsOverlay, NoFilteredRowsOverlay, NoMatchingRowsOverlay.

Reply:
My idea was NoResultOverlay

Reply:
I was also going to propose `NoFilteredRowsOverlay`.

Reply:
`NoResultsOverlay` could work too 👍

Reply:
Another suggestion: `NoVisibleRowsOverlay`. I know it doesn't make clear where it's used but it translates the condition used to display it. @DanailH do you want to help us decide?

Reply:
I'm in favour `NoResultsOverlay` since it sounds generic enough and we can use it in other situations later on but `NoVisibleRowsOverlay` is ok too.

Reply:
Changed to `NoResultsOverlay`.